zulipterminal/core.py

```diff
@@ -102,6 +102,15 @@ def raise_exception_in_main_thread(self,
             self._critical_exception = True
         os.write(self._exception_pipe, b'1')
 
+    def check_for_invalid_operation(self, stream_id: Optional[int]) -> bool:
+        assert stream_id is not None
+        if self.model.is_user_in_unsubscribed_stream(stream_id):
```

Comment:
The use of the assert here is confusing - why is it not just int?

I understand extracting the common code, but the function name seems very general.

Reply:
`stream_id` is `Optional[int]` currently, that's why the `assert` was necessary. When I first wrote this PR, `stream_id` which is actually `self.stream_id` was `int`. This function is called in 2 or 3 places, in each of them I had to put an `assert`, so thought of just pushing it in this function.

Reply:
I think `stream_id` should be `int` in the function declaration itself.